src/slurm_plugin/clustermgtd.py

```diff
@@ -689,54 +623,31 @@ def _handle_health_check(
         }
         log.info("Performing actions for health check type: %s", health_check_type)
         nodes_failing_health_check = []
+        failed_health_check_nodes_in_replacement = []
+        bootstrap_failure_nodes = []
         # Pick corresponding health check function
         is_instance_unhealthy = health_check_functions.get(health_check_type).get("function")
         default_kwargs = health_check_functions.get(health_check_type).get("default_kwargs")
         for instance in unhealthy_instance_status:
-            if is_instance_unhealthy(instance_health_state=instance, **default_kwargs):
-                unhealthy_node = ip_to_slurm_node_map.get(id_to_instance_map.get(instance.id).private_ip)
+            unhealthy_node = instance_id_to_active_node_map.get(instance.id)
+            if is_instance_unhealthy(instance, **default_kwargs):
```

Comment:
We probably don't need this `is_instance_unhealthy(instance, **default_kwargs)` function anymore, and we can do something like `instance_health_state.is_failing_scheduled_event or instance_health_state.is_failing_ec2_health` instead?